config/ssoadmin/config.go

```diff
@@ -0,0 +1,36 @@
+package ssoadmin
+
+import (
+	"github.com/crossplane/upjet/pkg/config"
+
+	"github.com/upbound/provider-aws/config/common"
+)
+
+// Configure adds configurations for the ssoadmin group.
+func Configure(p *config.Provider) {
+	p.AddResourceConfigurator("aws_ssoadmin_account_assignment", func(r *config.Resource) {
+		r.References["principal_id"] = config.Reference{
+			TerraformName:     "aws_identitystore_group",
+			Extractor:         `github.com/crossplane/upjet/pkg/resource.ExtractParamPath("group_id",true)`,
+			RefFieldName:      "PrincipalGroupRef",
+			SelectorFieldName: "PrincipalGroupSelector",
+		}
+		r.References["permission_set_arn"] = config.Reference{
+			TerraformName: "aws_ssoadmin_permission_set",
+			Extractor:     common.PathARNExtractor,
+		}
+	})
+	p.AddResourceConfigurator("aws_ssoadmin_customer_managed_policy_attachment", func(r *config.Resource) {
+		r.References["customer_managed_policy_reference.name"] = config.Reference{
+			TerraformName:     "aws_iam_policy",
+			RefFieldName:      "PolicyNameRef",
+			SelectorFieldName: "PolicyNameSelector",
+		}
+	})
+	p.AddResourceConfigurator("aws_ssoadmin_permission_set_inline_policy", func(r *config.Resource) {
+		r.References["instance_arn"] = config.Reference{
```

Comment:
I guess this reference should be defined for `permission_set_arn`

```suggestion
		r.References["permission_set_arn"] = config.Reference{
```

Reply:
This is extracting the `spec.forProvider.instanceArn` from the `PermissionSet` to use in the `PermissionSetInlinePolicy`. It does require repeating the same selector/reference twice in the managed resource manifest to do so, which is a bit awkward, and it's not obvious from the name that the `InstanceArnSelector` should actually point to a `PermissionSet`. It still seemed better than requiring the user to manually specify the instance arn an additional time. 

I didn't come up with this idea myself, actually. I noticed that upjet generated a reference identical to this one on one of the other resources, from the terraform example. 

Because PermissionSet is the one resource that is referenced by every other resource in this group, it made sense to me that we could use that as a source. Ultimately the "right" object to reference would be a terraform [data.aws_ssoadmin_instances](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ssoadmin_instances) resource, or perhaps a not-yet-existing resource version of that data source. 

I now notice that it's present on some but not all of the SSO resources, so I think it would be better to either add it where it's missing or take it off everywhere (including the places where it was automatically generated). I'm happy to do whichever you prefer.

Reply:
I ended up taking it off.